scripts/fdbshow

```diff
@@ -135,53 +137,80 @@ class FdbShow(object):
         return
     
     
-    def get_iter_index(self, key_value, pos=0):
-        """
-            Get the starting index of matched entry
-        """
-        if pos != 0:
-            self.bridge_mac_list = natsorted(self.bridge_mac_list, key = lambda x: x[pos])
-
-        keys = [r[pos] for r in self.bridge_mac_list]
-        return keys.index(key_value)
-
-
-    def display(self, vlan, port):
+    def display(self, vlan, port, address, entry_type, count):
         """
             Display the FDB entries for specified vlan/port.
             @todo: - PortChannel support
         """
         output = []
 
         if vlan is not None:
-            vlan = int(vlan)
-            s_index = self.get_iter_index(vlan)
-            self.bridge_mac_list = [fdb for fdb in self.bridge_mac_list[s_index:]
-                                    if fdb[0] == vlan]
-        if port is not None:
-            s_index = self.get_iter_index(port, 2)
-            self.bridge_mac_list = [fdb for fdb in self.bridge_mac_list[s_index:]
-                                    if fdb[2] == port]
+            vlan_val = int(vlan)
+
+        if address is not None:
+            address = address.upper()
 
-        for fdb in self.bridge_mac_list:
-            self.FDB_COUNT += 1
-            output.append([self.FDB_COUNT, fdb[0], fdb[1], fdb[2], fdb[3]])
+        if entry_type is not None:
+            entry_type = entry_type.capitalize()
 
-        print(tabulate(output, self.HEADER))
-        print("Total number of entries {0}".format(self.FDB_COUNT))
+        self.bridge_mac_list = [fdb for fdb in self.bridge_mac_list
+                                if (vlan is None or fdb[0] == vlan_val) and
+                                   (port is None or fdb[2] == port) and
+                                   (address is None or fdb[1] == address) and
+                                   (entry_type is None or fdb[3] == entry_type)]
 
+        if not count:
+            for fdb in self.bridge_mac_list:
+                self.FDB_COUNT += 1
```

Comment:
>self.FDB_COUNT [](http://example.com/codeflow?start=16&length=14)

Mixing class var and instance var is confusing. In this case, just use a function local var?

Reply:
Addressed.